src/main/scala/mesosphere/marathon/metrics/current/DropwizardMetrics.scala

```diff
@@ -0,0 +1,130 @@
+    DummyMetrics.deprecatedMinMaxCounter(prefix, `class`, metricName, tags, unit)
+
+  override def deprecatedTimer(prefix: MetricPrefix, `class`: Class[_], metricName: String,
+    tags: Map[String, String] = Map.empty, unit: Time = Time.Nanoseconds): Timer =
+    DummyMetrics.deprecatedTimer(prefix, `class`, metricName, tags, unit)
+
+  private val namePrefix = metricsConf.metricsNamePrefix()
+  private val histogramReservoirSignificantDigits = metricsConf.metricsHistogramReservoirSignificantDigits()
+  private val histogramReservoirResetPeriodically = metricsConf.metricsHistogramReservoirResetPeriodically()
+  private val histogramReservoirResettingIntervalMs = metricsConf.metricsHistogramReservoirResettingIntervalMs()
+  private val histogramReservoirResettingChunks = metricsConf.metricsHistogramReservoirResettingChunks()
+
+  implicit class DropwizardCounter(val counter: codahale.metrics.Counter) extends Counter {
+    override def increment(): Unit = increment(1L)
+    override def increment(times: Long): Unit = counter.inc(times)
+  }
+  def counter(name: String, unit: DropwizardUnitOfMeasurement = DropwizardUnitOfMeasurement.None): Counter = {
+    registry.counter(constructName(name, "counter", unit))
+  }
+
+  class DropwizardClosureGauge(val name: String, fn: () => Long) extends ClosureGauge {
+    registry.gauge(name, () => () => fn())
+  }
+  def closureGauge(name: String, fn: () => Long,
+    unit: DropwizardUnitOfMeasurement = DropwizardUnitOfMeasurement.None): ClosureGauge = {
+    new DropwizardClosureGauge(constructName(name, "gauge", unit), fn)
+  }
+
+  class DropwizardSettableGauge(val name: String) extends SettableGauge {
+    private[this] val register = new AtomicLong(0)
+    registry.gauge(name, () => () => register.get())
+
+    override def increment(by: Long): Unit = register.addAndGet(by)
+    override def decrement(by: Long): Unit = increment(-by)
+    override def value(): Long = register.get()
+    override def setValue(value: Long): Unit = register.set(value)
+  }
+  def gauge(name: String, unit: DropwizardUnitOfMeasurement = DropwizardUnitOfMeasurement.None): Gauge = {
+    new DropwizardSettableGauge(constructName(name, "gauge", unit))
+  }
+  def settableGauge(
+    name: String,
+    unit: DropwizardUnitOfMeasurement = DropwizardUnitOfMeasurement.None): SettableGauge = {
+    new DropwizardSettableGauge(constructName(name, "gauge", unit))
+  }
+
+  implicit class DropwizardTimerAdapter(val timer: metrics.Timer) extends TimerAdapter {
+    override def update(duration: FiniteDuration): Unit = timer.update(duration)
+    override def update(value: Long): Unit = timer.update(value, TimeUnit.NANOSECONDS)
+  }
+
+  private val timersLock = RichLock()
+  def timer(name: String): Timer = timersLock {
+    val effectiveName = constructName(name, "timer", DropwizardUnitOfMeasurement.Time)
+    if (registry.getTimers().containsKey(effectiveName)) {
```

Comment:
Why don't we check for existing metrics with e.g. _gauges_ or _counters_? What happens if you register >1 metric with the same name? And if nothing bad happens, maybe we don't need to synchronize here at all?

Reply:
Because we use `.register` only for timers. In the cases this check is done by Dropwizard itself. If `.register` is called twice with the same metric name, the second call will result into an exception thrown.